sphinx_immaterial/css_and_javascript_bundles.py

```diff
@@ -0,0 +1,199 @@
+    #
+    # https://sourcemaps.info/spec.html
+    sourcemap_sections = []
+    lines: List[str] = []
+
+    entries.sort(key=lambda entry: entry.priority)
+    for entry in entries:
+        entry_lines = [
+            x
+            for x in entry.code.rstrip().splitlines()
+            if not x.startswith(source_mapping_url_prefix)
+        ]
+        offset = len(lines)
+        lines.extend(entry_lines)
+        if entry.sourcemap:
+            sourcemap = json.loads(entry.sourcemap)
+            sourcemap_sections.append(
+                {"offset": {"line": offset, "column": 0}, "map": sourcemap}
+            )
+    lines.append("")
+    output_data = "\n".join(lines).encode("utf-8")
+
+    # hash data and write to generated file
+    output_data_hash = hashlib.sha256(output_data).hexdigest()
+    output_path = f"{output_prefix}.{output_data_hash[:17]}.min.{output_ext}"
+    output_path_obj = pathlib.Path(output_path)
+    output_path_obj.parent.mkdir(parents=True, exist_ok=True)
+    if sourcemap_sections and getattr(app.config, _BUNDLE_SOURCE_MAPS_KEY):
+        sourcemap_path = output_path + ".map"
+        pathlib.Path(sourcemap_path).write_text(
+            json.dumps({"version": 3, "sections": sourcemap_sections}),
+            encoding="utf-8",
+        )
+        output_data += (
+            source_mapping_url_prefix
+            + os.path.basename(sourcemap_path)
+            + source_mapping_url_suffix
+            + "\n"
+        ).encode("utf-8")
+    output_path_obj.write_bytes(output_data)
+    return output_path
+
+
+def generate_bundles(
+    app: sphinx.application.Sphinx, env: sphinx.environment.BuildEnvironment
+) -> None:
+    """Bundles the theme CSS with any additional CSS added using `add_global_css`."""
+
+    css_entries = list(_get_css_entries(app))
+
+    # copy pre-minified CSS from theme's bundles
+    theme_bundles = pathlib.Path(__file__).parent / "bundles" / "stylesheets"
+    css_entries.append(
+        Entry(
+            code=(theme_bundles / "main.css").read_text(encoding="utf-8"),
+            sourcemap=(theme_bundles / "main.css.map").read_text(encoding="utf-8"),
+            priority=200,
+        )
+    )
+
+    theme_options = app.config["html_theme_options"]
+    if theme_options and "palette" in theme_options:
+        css_entries.append(
+            Entry(
+                code=(theme_bundles / "palette.css").read_text(encoding="utf-8"),
+                sourcemap=(theme_bundles / "palette.css.map").read_text(
+                    encoding="utf-8"
+                ),
+                priority=200,
+            )
+        )
+
+    css_bundle = generate_bundle(
+        app=app,
+        env=env,
+        entries=css_entries,
+        source_mapping_url_prefix="/*# sourceMappingURL=",
+        source_mapping_url_suffix=" */",
+        output_prefix=os.path.join(app.outdir, "_static", "sphinx_immaterial_theme"),
+        output_ext="css",
+    )
+    app.add_css_file(css_bundle, priority=200)
```

Comment:
I think the resulting CSS path needs to be made relative to the `app.outdir`. The [RTD build looks broken](https://sphinx-immaterial--205.org.readthedocs.build/en/205/), but the CSS loads if I change the path (in the live DOM) to be relative.
```diff
-<link rel="stylesheet" type="text/css" href="/home/docs/checkouts/readthedocs.org/user_builds/sphinx-immaterial/checkouts/205/docs/_build/html/_static/sphinx_immaterial_theme.b26c02e5c3d38b40b.min.css">
+<link rel="stylesheet" type="text/css" href="_static/sphinx_immaterial_theme.b26c02e5c3d38b40b.min.css">
```

Reply:
Thanks for catching that --- should be fixed now.

Reply:
Indeed, it is fixed.